src/Classes/Daemons/MyRadio_TrackAndTraceDaemon.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace MyRadio\Daemons;
+
+use MyRadio\Config;
+use MyRadio\ServiceAPI\MyRadio_Timeslot;
+use MyRadio\ServiceAPI\MyRadio_Season;
+use MyRadio\MyRadio\CoreUtils;
+use MyRadio\MyRadioEmail;
+use MyRadio\ServiceAPI\MyRadio_List;
+
+class MyRadio_TrackAndTraceDaemon extends \MyRadio\MyRadio\MyRadio_Daemon
+{
+    public static function isEnabled()
+    {
+        return Config::$d_TrackAndTrace_enabled;
+    }
+
+    public static function run()
+    {
+        $weekkey = __CLASS__ . '_last_run_weekly';
+        if (self::getVal($weekkey) > time() - 604700) {
+            return;
+        }
+
+        self::generateTrackAndTraceReport();
+
+        // Done
+        self::setVal($weekkey, time());
+    }
+
+    private static function generateTrackAndTraceReport()
+    {
+        $table = "<table>";
+        $table .= "<tr><th>Type</th><th>Information</th><th>Location</th><th>Time</th></tr>";
+
+        $data = [];
+        $no_track = MyRadio_Timeslot::getLocationName(5); //WebStudio
+
+        foreach (MyRadio_Season::getAllSeasonsInLatestTerm() as $season) {
+            foreach ($season->getAllTimeslots() as $timeslot) {
+                if ($timeslot->getStartTime() < time()
+                    && $timeslot->getStartTime() > time() - 604800
+                ) {
+                    foreach ($timeslot->getSigninInfo() as $info) {
+                        if (isset($info["location"]) && $info["location"] != $no_track) {
+                            if (isset($info["user"])) {
+                                $eduroam = $info["user"]->getEduroam();
```

Comment:
People without eduroams exist. Idk if any will want to use the studios though, but is there a reason you used getEduroam instead of getEmail?

Reply:
some people have personal emails set, but cause yusu/uni we need eduroams.